File: 2022/OpenCV4 tutorial/stitching/stitch.py
import sys
import cv2 as cv

if __name__ == "__main__":
    stitcher = cv.Stitcher_create(cv.STITCHER_PANORAMA)
    panos = []
    panos.append(cv.imread("1403636579763555584cam0.png"))
    panos.append(cv.imread("1403636579763555584cam1.png"))
    status, out = stitcher.stitch(panos)
    if status != cv.STITCHER_OK:
        print(f'Can\'t stitch images, error code = {status}')
        sys.exit(1)
    cv.imshow("out", out)
    cv.waitKey()
    cv.destroyAllWindows()

File: 2022/OpenCV4 tutorial/stitching/stitch.cs
using System;
using System.Collections.Generic;
using System.IO;
using OpenCvSharp;

namespace Stitching
{
    class Program
    {

        static void Main(string[] args)
        {
            DirectoryInfo rootDir = Directory.GetParent(Environment.CurrentDirectory);
            string root = rootDir.Parent.Parent.FullName;
            System.IO.Directory.SetCurrentDirectory(root);


            Mat outImg = new Mat();
            IList<Mat> panos = new List<Mat>();
            panos.Add(Cv2.ImRead(@".\1403636579763555584cam0.png"));
            panos.Add(Cv2.ImRead(@".\1403636579763555584cam1.png"));
            Stitcher sticher = Stitcher.Create(Stitcher.Mode.Panorama);
            Stitcher.Status status = sticher.Stitch(panos, outImg);
            if (status != Stitcher.Status.OK)
            {
                Console.WriteLine("Can't stitch images, error code = {0}", ((int)status));
                return;
            }
            Cv2.ImShow("out", outImg);
            Cv2.WaitKey();

        }
    }
}
